elegant/process_experiment.py

```diff
@@ -80,11 +81,23 @@ def segment_experiment(experiment_root, model, channels='bf', use_gpu=True, over
         overwrite_existing: if False, the segmenter will not be run on existing
             mask files, nor will existing annotations be modified even if new
             mask files are generated for a timepoint.
+        timepoint_filter - filter for scan_experiment_dir specifying timepoints to segment
+        mask_root - root directory for masks that will be used to updating annotations
     """
+    if mask_root is None:
+        mask_root = experiment_root / 'derived_data' / 'mask'
+
     experiment_root = pathlib.Path(experiment_root)
-    positions = load_data.scan_experiment_dir(experiment_root, channels=channels)
-    mask_root = experiment_root / 'derived_data' / 'mask'
-    segment_images.segment_positions(positions, model, mask_root, use_gpu, overwrite_existing)
+    positions = load_data.scan_experiment_dir(experiment_root, channels=channels, timepoint_filter=timepoint_filter)
+    if positions:
+        process = segment_images.segment_positions(positions, model, mask_root, use_gpu, overwrite_existing)
+        if process.stderr:
+            print(f'Errors during segmentation: {process.stderr}')
+        with (mask_root / 'notes.txt').open('a+') as notes_file:
+            notes_file.write(
+                f'{datetime.datetime.today().strftime("%Y-%m-%dt%H%M")} These masks were segmented with model {model}\n')
```

Comment:
This seems like pretty useless logging. Which masks do "these" refer to?If you want to keep a log, have the log (a) be easy to machine-parse, and (b) clearly map date/time and model file to each mask file generated. Ideally one line per mask file to make everything dead simple. (This also suggests separating out an error log from the success log.) Last, calling a logfile `notes.txt` is not maximally descriptive. Even `segmentation_log` would be more useful.

Reply:
Fair points. Obviously, better file naming and more systematic, machine-parsable content are attainable.

Currently, generating the relevant files for segmenting and actually doing the segmentation are separated over the Python/executable divide. As far as mapping date+time+model to mask files, it's not clear how to create this mapping, particularly in the case where an error causes premature stoppage of segmentation. Maybe append process.stdout? A little dirty, but would contain all the relevant information generated while running the executable. What do you think?